inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -171,7 +171,7 @@ def _rollback_and_delete_snap(self) -> None:
         a.) After reboot by SOTA, and diagnostic reports bad report for system health
         """
         logger.debug("")
-        if self.snap_num:
+        if self.snap_num and self.snap_num is not "0":
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Maintainability$</h3>

Consider using `!=` instead of `is not` for comparing `self.snap_num` with the string "0" to avoid potential issues with string identity checks.

Reply:
@tsirlapu "is not" and "!=" do different things in Python--idiomatic would be if they did the same thing but one was more familiar. see https://www.geeksforgeeks.org/difference-between-and-is-not-operator-in-python/